client/extensions/woocommerce/state/sites/settings/email/reducer.js

```diff
@@ -347,6 +348,27 @@ function newsletterSettingsSubmitError( state = false, action ) {
 	return state;
 }
 
+/**
+ * Updates state with save settings request status
+ * The name ( saveSettings ) is generic because in future this should repond
+ * also to other events than newsletter settings
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action
+ * @return {Object}        Updated state
+ */
+function saveSettings( state = false, { type } ) {
+	switch ( type ) {
+		case WOOCOMMERCE_MAILCHIMP_SAVE_SETTINGS:
+			return true;
+		case WOOCOMMERCE_MAILCHIMP_NEWSLETTER_SETTINGS_SUBMIT_SUCCESS:
+		case WOOCOMMERCE_MAILCHIMP_NEWSLETTER_SETTINGS_SUBMIT_FAILURE:
```

Comment:
Why just the newsletter (response) action types here? Why not the others too (i.e. if you change the API key, etc etc)

Reply:
This is the only one that requires this. There is a chance that in future we will want to extend this but for now only this is needed.

Reply:
Would it make sense then to just use the mailchimpNewsletterSettingsSubmit action creator, adding a flag to state for that and not bother adding this one?

Reply:
Yes. Branching of a branch is a useful mental concept for the person who works on that part of code, but it can be very confusing and frustrating for the reviewers :/ I wonder if there is a way to help reviewers - probably good end excessive description in PR is the best way.